platform/server/http/Proxy/Proxy.ts

```diff
@@ -0,0 +1,84 @@
+import { Server } from 'net';
+import { IncomingMessage, OutgoingMessage } from 'http';
+import { EventEmitter } from 'events';
+
+import { Logger } from '../../../logging';
+
+export class Proxy extends EventEmitter {
+  private server: Server;
+  private readonly eventHandlers: any;
+  private port: number;
+
+  constructor(private readonly log: Logger, readonly kbnServer: any) {
+    super();
+
+    const forwardEvent = (name: string, ...args: any[]) => {
+      this.log.info(`Event is being forwarded: ${name}`);
+      this.emit(name, ...args);
+    };
+
+    this.eventHandlers = {
+      listening: forwardEvent.bind(this, 'listening'),
+      error: forwardEvent.bind(this, 'error'),
+      clientError: forwardEvent.bind(this, 'clientError'),
+      connection: forwardEvent.bind(this, 'connection')
+    };
+  }
+
+  address() {
+    return this.server && this.server.address();
+  }
+
+  listen(port: number, host: string, callback: Function) {
+    this.log.info(`"listen" has been called (${host}:${port}).`);
+    this.port = port;
+
+    this.emit('platform-start');
+    callback && callback();
```

Comment:
Ideally we should call `callback` only when platform is ready, but I should think how to do that. For now I haven't noticed any issues with this approach.

Reply:
If we removed the events and rather injected `start` and `stop` functions when creating the `Proxy`, we could call the `callback` when they finish instead.

Reply:
Good idea ✔️